src/initdt.jl

```diff
@@ -35,14 +35,14 @@ function ode_determine_initdt{tType,uType}(u0,t::tType,tdir,dtmax,abstol,reltol,
 
   #@. u₁ = @muladd u0 + tdir*dt₀*f₀
   @tight_loop_macros for i in uidx
-    @inbounds u₁[i] = u0[i] + tdir*dt₀*f₀[i]
+    @inbounds u₁[i] = u0[i] + (tdir*dt₀)*f₀[i]
```

Comment:
Yes that's probably the best. `@muladd` works on this now, it handles also products with more than two factors (should write a test for it...). In the current implementation `@muladd a*b*c+d` is transformed to `muladd(a, b*c, d)` - so the first factor always ends up as the first argument to `muladd` and the product of all other factors builds the second argument. I don't know if this is better/more natural than `muladd(a*b, c, d)`.

In this case `@muladd` produces
```julia
julia> macroexpand(:(@muladd @tight_loop_macros for i in uidx
           @inbounds u₁[i] = u0[i] + (tdir*dt₀)*f₀[i]
       end))
:(for i = uidx # REPL[6], line 2:
        begin 
            $(Expr(:inbounds, true))
            u₁[i] = (muladd)(tdir * dt₀, f₀[i], u0[i])
            $(Expr(:inbounds, :pop))
        end
    end)
```
and without brackets:
```julia
julia> macroexpand(:(@muladd @tight_loop_macros for i in uidx
           @inbounds u₁[i] = u0[i] + tdir*dt₀*f₀[i]
       end))
:(for i = uidx # REPL[7], line 2:
        begin 
            $(Expr(:inbounds, true))
            u₁[i] = (muladd)(tdir, dt₀ * f₀[i], u0[i])
            $(Expr(:inbounds, :pop))
        end
    end)
```
But of course the best is to move the multiplication of the first two factors completely out of the loop.